### PR TITLE
progress all support successPercent

### LINKED ISSUE #14332: Circular progress bar should support success segment
- [X] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
Linear Progress Bars have the option to add a secondary success segment embedded. This feature, however, is not supported in circular or dashboard Progress, as seen in [this fiddle](https://codesandbox.io/s/vmn6zowxv5)

### What does the proposed API look like?
`<Progress percent={60} successPercent={30} type="circle" />` should work similarly as to how `<Progress percent={60} successPercent={30} type="line" />` does

### PR DESCRIPTION
**Pull request by zombieJ**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch.
Pull request will be merged after one of collaborators approve.
Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

close #14332
  
### API Realization (Optional if not new feature)

```jsx
<Progress percent={60} successPercent={30} />
<Progress percent={60} successPercent={30} type="circle" />
<Progress percent={60} successPercent={30} type="dashboard" />
```

### What's the effect? (Optional if not new feature)

No affects.

### Changelog description (Optional if not new feature)

1. Progress all types support `successPercent` prop.
2. Progress 所有类型都支持 `successPercent` 属性。

### Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/progress/demo/segment.md [hunk 1]
```diff
@@ -19,6 +19,10 @@ import { Tooltip, Progress } from 'antd';
 ReactDOM.render(
   <Tooltip title="3 done / 3 in progress / 4 to do">
     <Progress percent={60} successPercent={30} />
+
+    <Progress percent={60} successPercent={30} type="circle" />
+
+    <Progress percent={60} successPercent={30} type="dashboard" />
```
Comment (afc163) [inline components/progress/demo/segment.md]: 应该放在各自的 Tooltip 里。

Comment (zombieJ) [reply components/progress/demo/segment.md]: OPS...疏忽

File: components/progress/progress.tsx [hunk 2]
```diff
@@ -155,13 +155,24 @@ export default class Progress extends React.Component<ProgressProps, {}> {
...
+        const successPercentVal = validProgress(successPercent);
+        percents = [successPercentVal, percents - successPercentVal];
+        strokeColors = [statusColorMap.success, strokeColors];
+      }
+
       progress = (
         <div className={`${prefixCls}-inner`} style={circleStyle}>
           <Circle
-            percent={validProgress(percent)}
+            percent={percents}
```
Comment (afc163) [inline components/progress/progress.tsx]: 搞个 `this.getPercents()` 和 `this.getStrokeColors()` 吧， render 里东西有点多了。

Comment (zombieJ) [reply components/progress/progress.tsx]: 我们的 line 是自己做的，不是来自 rc-progress。我抽一个 renderLine 和 renderCircle 吧~

**Review by ztplz — DISMISSED** [review]

File: components/progress/progress.tsx (hunk 2 above)
Comment (afc163) [reply components/progress/progress.tsx]: 我意思是这个 class 里的代码梳理一下就行了。

Comment (ztplz) [reply components/progress/progress.tsx]: 可以把 line 的代码也移到rc那边去啊

Comment (zombieJ) [reply components/progress/progress.tsx]: @ztplz rc 那边有个 line，这里没有用。不清楚历史原因是啥，不过 rc 那边 stackValue 两个图我都实现好了。

**Comment by zombieJ** [comment]
done. Check again. @afc163 @ztplz

File: components/progress/progress.tsx [hunk 3]
```diff
@@ -70,6 +72,87 @@ export default class Progress extends React.Component<ProgressProps, {}> {
...
+    if (successPercent) {
+      const successPercentVal = validProgress(successPercent);
+      percents = [successPercentVal, percents - successPercentVal];
+      strokeColors = [statusColorMap.success, strokeColors];
+    }
+
+    return (
+      <div className={`${prefixCls}-inner`} style={circleStyle}>
+        <Circle
+          percent={percents}
```
Comment (afc163) [inline components/progress/progress.tsx]: 好像没啥变化。



搞个 this.getPercents() 和 this.getStrokeColors() 吧， 这个 renderCircle 里东西有点多了。

Comment (afc163) [reply components/progress/progress.tsx]: 就是把这两个 render 再拆碎一点。

**Review by afc163 — APPROVED** [review]

**Pull request merged**